lib/collection_document.rb

```diff
@@ -1,36 +1,44 @@
 # frozen_string_literal: true
 
 module JekyllRPG
-  # Represents a document that may be in a Jekyll collection
+  # Represents a document that may be in a Jekyll collection,
+  # extracted from either a document itself, or a markdown
+  # link.
   class CollectionDocument
-    attr_accessor :name, :collection, :slug, :written
+    attr_accessor :name, :collection, :slug, :viewable
 
     def extract_doc(doc)
       @name = doc.data['name']
       @collection = doc.collection.label
       @slug = doc.data['slug']
-      @written = true
+      @viewable = true
       self
     end
 
     # extracts link text, collection and slug
-    # [@name](/@collection/@slug)
     def extract_markdown(site, link)
-      @collection = link[%r{(?<=/).*(?=/)}]
-      @slug = link[%r{(?<=/)(?:(?!/).)*?(?=\))}]
-      @written = document_exists(site)
-      @name = @written ? find_document(site).data['name'] : link[/(?<=\[).*?(?=\])/]
+      @site = site
+      @collection = link.collection
+      @slug = link.slug
+      @viewable = viewable
+      @name = @viewable ? find_document.data['name'] : link.name
       self
     end
 
     # Checks whether document exists in a site
-    def document_exists(site)
-      !site.collections[@collection].nil? && !find_document(site).nil?
+    def document_exists
+      !@site.collections[@collection].nil? && !find_document.nil?
```

Comment:
I think this could be expressed better as a positive assertion, e.g.
```suggestion
      @site.collections[@collection].present? || find_document.present?
```